airflow/models/all_models.py

```diff
@@ -0,0 +1,46 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Airflow models"""
+from airflow.models.base import ID_LEN, Base  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.baseoperator import (  # noqa: F401 # pylint: disable=unused-import
+    BaseOperator, BaseOperatorLink,
+)
+from airflow.models.connection import Connection  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.dag import DAG, DagModel, DagTag  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.dagbag import DagBag  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.dagpickle import DagPickle  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.dagrun import DagRun  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.errors import ImportError  # pylint: disable=redefined-builtin
+from airflow.models.log import Log  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.pool import Pool  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.skipmixin import SkipMixin  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.slamiss import SlaMiss  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.taskfail import TaskFail  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.taskinstance import (  # noqa: F401 # pylint: disable=unused-import
+    TaskInstance, clear_task_instances,
+)
+from airflow.models.taskreschedule import TaskReschedule  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.variable import Variable  # noqa: F401 # pylint: disable=unused-import
+from airflow.models.xcom import XCOM_RETURN_KEY, XCom  # noqa: F401 # pylint: disable=unused-import
+
+try:
+    from airflow.models.kubernetes import (  # noqa: F401 # pylint: disable=unused-import
+        KubeResourceVersion, KubeWorkerIdentifier
+    )
+except ImportError:
+    pass
```

Comment:
I wondering if we can create a `import_models()` function that will import all SQLA models using strings? We are doing it already for executors:
https://github.com/apache/airflow/blob/e936159672539b13ddecfe66aee1094fe8923291/airflow/executors/executor_loader.py#L41-L47

Reply:
Few more changes are needed here. I think we have not achieved yet what we wanted with lazy loading all models only when needed. 

The only reason why we are loading all models here is the "resetdb()" command. It's the only one that requires all models to be imported. So we need to get rid of all the model imports from the top-level db.py. This can be done with those changes:

1) make all_models a package rather than module and move all the imports to `__init__.py` and add __all__= 
'ID_LEN', 'BaseOperator', ....]  there . This way we could import all the models by "from airflow.models.all_models import *' - this is exactly what __init__.py and __all__ were designed for I think :).

2) Also move BaseJob and SerializedDagModel the all_models/__init__.py there (and remove them from db.py - this way we will achieve what we were supposed to do here - have one place

3) Then turn "from airflow.models.all_models import *" into a local import in the resetdb() method. Again - this is the only method that needs to import all models, so moving it as local import is a good idea for three reasons: 
 - when importing db.py we will not import all models immediately
 - it will be explicit that it's only the resetdb method needs those imports.
 - this is how we avoid cyclic imports from many places.